Max height drawing tool

srallen opened the pull request:
A variant of the rectangle drawing tool that is similar to the planet hunters style tool where the rectangle is the full height of the subject, but has adjustable width. This tool is requested for the Intro2Astro activity 4 where students will be classifying an area on a plot. 

You can see it in action with the intended subjects here (with pan and zoom also enabled):
https://max-height-drawing-tool.pfe-preview.zooniverse.org/projects/srallen086/intro2astro-testing/classify

Currently it is also an experimental feature. My hope is to get this merged in next week and to get the project setup for the class to use on April 28.

brian-c commented on app/classifier/drawing-tools/column-rectangle.cjsx:
```diff
@@ -0,0 +1,81 @@
+
+    defaultValues: ({x, y}) ->
+      x: x
+      y: 0
```
Will `y` ever not be `0`? Why record it?

brian-c commented on app/classifier/drawing-tools/column-rectangle.cjsx:
```diff
@@ -0,0 +1,81 @@
+
+  render: ->
+    {x, y, width} = @props.mark
+    allowedWidth = if @props.mark.width < MINIMUM_WIDTH then MINIMUM_WIDTH else width
```
This decouples the display of the mark from the values in the annotation. Enforce the minimum in the collected data, not the rendered output.

brian-c commented on app/classifier/drawing-tools/column-rectangle.cjsx:
```diff
@@ -0,0 +1,81 @@
+Draggable = require '../../lib/draggable'
+DeleteButton = require './delete-button'
+
+DELETE_BUTTON_DISTANCE = 9 / 10
```
This is unused.

brian-c commented on app/classifier/drawing-tools/column-rectangle.cjsx:
```diff
@@ -0,0 +1,81 @@
+    @props.mark.width += d.x / @props.scale.horizontal
+    @props.onChange @props.mark 
+
+  normalizeMark: ->
```
Assuming you're going of the normal rectangle tool: the normal rectangle renders as a `<polyline>` in order the handle negative dimensions while a drag is in progress. We're rendering a `<rect>` here, so I don't think this make too much sense.

brian-c commented:
A couple overall things:
- [ ] Seems like there's some horsiness when dragging the handles (although not on initial drag) where the recorded coordinates get out of sync with the mouse.
- [ ] Implement `deleteIfOutOfBounds` on the main Draggable.
- [ ] Add an icon in classifier/tasks/drawing/icons
- [ ] Add tool to the drawing task in page/dev-classifier/mock-data

Or if you need this merged pronto for a demo, say the word. Works mostly fine!

srallen commented:
Thanks. I'll address what I can today, but will need it merged sometime early next week. There will only be one pilot class at NU using it in a project, so some of this may need to come in a later PR.

brian-c commented:
Sounds good. The data/render out-of-sync comment is the only really scary one.

srallen commented:
@brian-c I've removed the unnecessary or unused code and reworked the min-width to be in the handlers. I'm still seeing some odd UX behavior if you drag a handle beyond the min-width either direction and back again, but it the annotated mark width and what is rendered seems in sync now.

Gonna start to take a look at some of the other suggestions now.

srallen commented:
Got the other points done now. This is ready for review again.

brian-c commented:
Solid, thanks!
